dianna/methods/shap.py

```diff
@@ -1,21 +1,223 @@
+                                     interpretability, training a separate interpretable
+                                     model to explain a model’s prediction on each individual
+                                     example. The input dimension must be
+                                     [height, width, color_channels] or
+                                     [color_channels, height, width] (see channel_axis_first)
+            nsamples ("auto" or int): Number of times to re-evaluate the model when
+                                      explaining each prediction. More samples lead
+                                      to lower variance estimates of the SHAP values.
+                                      The "auto" setting uses
+                                      `nsamples = 2 * X.shape[1] + 2048`
+            background (int): Background color for the masked image
+            n_segments (int): The (approximate) number of labels in the segmented output image
+            compactness (int): Balances color proximity and space proximity. Higher values give
+                               more weight to space proximity, making superpixel shapes more
+                               square/cubic.
+            sigma (float): Width of Gaussian smoothing kernel for pre-processing for
+                           each dimension of the image. Zero means no smoothing.
+            channel_axis_first (boolean): If True, the shape of the input image is
+                                          [color_channels, height, width]
+
+        Other keyword arguments: see the documentation of kernel explainer of SHAP
+                                 (also in function "shap_values") via:
+        https://github.com/slundberg/shap/blob/master/shap/explainers/_kernel.py
+                                 and the documentation of image segmentation via:
+        https://scikit-image.org/docs/dev/api/skimage.segmentation.html#skimage.segmentation.slic
+
+        Returns:
+            Explanation heatmap of shapley values for each class (np.ndarray).
         """
-        Example call function.
+        self.onnx_model = onnx.load(model)  # load onnx model
+        self.output_node = prepare(self.onnx_model, gen_tensor_dict=True).outputs[0]
+        self.input_data = input_data
+        self.channel_axis_first = channel_axis_first
+        self.background = background
+        # other keyword arguments for the method segment_image
+        max_num_iter = kwargs.get("max_num_iter", 10)
+        spacing = kwargs.get("spacing", None)
+        convert2lab = kwargs.get("convert2lab", None)
+        enforce_connectivity = kwargs.get("enforce_connectivity", True)
+        min_size_factor = kwargs.get("min_size_factor", 0.5)
+        max_size_factor = kwargs.get("max_size_factor", 3)
+        slic_zero = kwargs.get("slic_zero", False)
+        start_label = kwargs.get("start_label", 1)
+        mask = kwargs.get("mask", None)
```

Comment:
All these are unnecessary if you just pass `kwargs`. The default values are also the same as `slic`'s, right? So those too can be left out in DIANNA. To determine which kwargs can be passed to `segment_image` (or rather `slic`), you can use `slic_kwargs = dianna.utils.get_kwargs_applicable_to_function(slic, kwargs)` (see here: https://github.com/dianna-ai/dianna/blob/main/dianna/utils/misc.py#L26).

Reply:
Thanks @egpbos for providing the solution! Indeed I was a bit confused about how to pick up the right argument for the `slic` function as we have already passed `kwargs` to the class when creating the object. I didn't notice that we've already got `dianna.utils.get_kwargs_applicable_to_function(slic, kwargs)`. I have removed those unnecessary arguments and now the code looks much much cleaner! Thanks agin. A big thumb up!!!